### PR TITLE
Feature: Use currency control for goal amount

### PR DESCRIPTION
**Pull request by JasonTheAdams**
## Description

This simple little PR uses the Currency control for the Goal amount, which was previously a number control. So it looks all nice, now! I also increased the debounce delay to 500ms, as 100ms just felt way too short.

## Affects

The donation goal

## Visuals

image

## Testing Instructions

Play around with the donation goal and make sure it looks right and works in the front-end.

## Pre-review Checklist

-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205076251265772
  - https://app.asana.com/0/0/1205681940003053

### CODE REVIEW

File: src/FormBuilder/resources/js/form-builder/src/settings/donation-goal/index.tsx [hunk 1]
```diff
@@ -69,11 +69,11 @@ const DonationGoalSettings = () => {
                         />
                     </PanelRow>
                     <PanelRow>
-                        <NumberControl
+                        <CurrencyControl
```
Comment (jonwaldstein) [inline src/FormBuilder/resources/js/form-builder/src/settings/donation-goal/index.tsx:72]: @JasonTheAdams the goal amount is an arbitrary number that is used throughout every goal type.  In order for a currency control to make sense, you would need to first make sure the goal type refers to a dollar amount.

**Review by jonwaldstein — CHANGES_REQUESTED** [review]

File: src/FormBuilder/resources/js/form-builder/src/settings/donation-goal/index.tsx (hunk 1 above)
Comment (JasonTheAdams) [reply src/FormBuilder/resources/js/form-builder/src/settings/donation-goal/index.tsx:72]: Hah! Oh yeah. Good catch!

Comment (JasonTheAdams) [reply src/FormBuilder/resources/js/form-builder/src/settings/donation-goal/index.tsx]: Resolved in 5e8217d

**Review by jonwaldstein — APPROVED** [review]
Looks good now 👍

**Review by rickalday — APPROVED** [review]
Passed manual QA tests.

**Pull request merged**